server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,218 @@
+      onDrop={this.onDrop.bind(this)}
+      style={{height: this.props.baseCanvas.style.height, width: this.props.baseCanvas.style.width}}>
+      {textToolBox}
+    </div>;
+  }
+
+  renderTextBox() {
+    const dragDivTopPx = `${this.state.top - DRAG_PADDING}px`;
+    const dragDivLeftPx = `${this.state.left - DRAG_PADDING}px`;
+    const dragDivPadding = `${DRAG_PADDING}px`;
+
+    const dragDivStyles = {
+      position: "absolute",
+      top: dragDivTopPx,
+      left: dragDivLeftPx,
+      padding: dragDivPadding,
+      cursor: "move"
+    };
+
+    return [
+      <canvas ref={this.canvas} id="text-canvas" key="base" width={this.props.baseCanvas.width}
+        height={this.props.baseCanvas.height}
+        style={{width: this.canvasCssWidth,
+                height: this.canvasCssHeight}}>
+      </canvas>,
+
+      <div draggable="true" key="drag" onDragStart={this.onDragStart.bind(this)} style={dragDivStyles}>
+        <div id="text-input" ref={this.textInput} contentEditable="true" key="text" onKeyUp={this.drawText.bind(this)}
+           className={`${this.state.textSize} ${this.state.colorName} text`} >
+        </div>
+      </div>
+    ];
+  }
+
+  renderToolbar() {
+    return <div className="editor-header default-color-scheme"><div className="annotation-tools">
+      <Localized id="annotationTextSize">
+        <select className={`text-select`} title="Text Size" onChange={this.onChangeTextSize.bind(this)} value={this.state.textSize}>
+          <Localized id="textSizeSmall"><option value="small-text">Small</option></Localized>
+          <Localized id="textSizeMedium"><option value="medium-text">Medium</option></Localized>
+          <Localized id="textSizeLarge"><option value="large-text">Large</option></Localized>
+        </select>
+      </Localized>
+      <ColorPicker activeTool={this.state.tool}
+        setColorCallback={this.setColor.bind(this)}
+        color={this.state.color} />
+      <span className="annotation-divider"></span>
+      <Localized id="annotationTextConfirmButton">
+        <button className={`button transparent confirm-text`} id="confirm-text" onClick={this.onClickConfirm.bind(this)} title="Confirm Text">Confirm</button>
+      </Localized>
+      <Localized id="annotationTextCancelButton">
+        <button className={`button transparent cancel-text`} id="cancel-text" onClick={this.onClickCancel.bind(this)} title="Cancel Text">Cancel</button>
+      </Localized>
+    </div></div>;
+  }
+
+  setColor(color, colorName) {
+    this.setState({color});
+    this.setState({colorName});
+    this.textInput.current.focus();
+  }
+
+  drawText(e) {
+    this.setState({textContent: e.target.textContent});
+  }
+
+  onClickConfirm(e) {
+    const styles = window.getComputedStyle(this.textInput.current);
+
+    const BKG_COLOR = styles["background-color"];
+    const FONT_COLOR = styles.color;
+    const FONT_SIZE = parseInt(styles["font-size"].replace(/px/i, ""), 10);
```

Comment:
The regex is unnecessary.  `parseInt` will return the correct value.

Reply:
+1